### ISSUE TITLE
ansible ec2_elb_facts doesn't use the credentials file

### ISSUE THREAD

**Issue by coffeepac** [samsung-cnct/k2#628]
according to this page:  http://docs.ansible.com/ansible/ec2_elb_facts_module.html there is no option to use the aws credentials file.  We allow a user to specify only a credentials file, which would then not get used for querying the ELBs.

I am somewhat at a loss as to how this ever worked unless there is something in K2 that exports the values in the credentials file into the environment?

**Comment by davidewatson** [comment]
Huh. This should be fixed.

Ansible uses the boto configuration file if no credentials are provided. This includes `~/.aws` et al. The [documentation](http://boto.readthedocs.io/en/latest/boto_config_tut.html) contains more.

Notes
------

The default config:

```
      authentication:
        accessKey:
        accessSecret:
        credentialsFile: "$HOME/.aws/credentials"
        credentialsProfile:
```

The kraken.ssh role which depends on it:

```
- name: Gather inventory of all cluster nodes
  ec2_remote_facts:
    region: "{{ cluster.providerConfig.region }}"
    aws_access_key: "{{ cluster.providerConfig.authentication.accessKey or omit }}"
    aws_secret_key: "{{ cluster.providerConfig.authentication.accessSecret or omit }}"
    profile: "{{ cluster.providerConfig.authentication.credentialsProfile or omit }}"
    filters:
      "tag:k2-nodepool": "{{ item.name }}"
      "tag:Name": "{{ cluster.name }}_{{ item.name }}*"
  with_items: "{{ cluster.nodePools }}"
  register: ec2_results
```

**Comment by coffeepac** [comment]
that only works if the credentials file is in a users HOME directory

**Comment by davidewatson** [comment]
Right. The only place where the credentialsFile config variable is referenced in k2 is the `kraken.provider.aws` role. It will allow you to create terraform resources. Stuff that breaks includes the `kraken.ssh` role.

```
bash-4.3# grep -r credentialsFile ansible
ansible/roles/kraken.config/files/config.yaml:        credentialsFile: "$HOME/.aws/credentials"
ansible/roles/kraken.provider/kraken.provider.aws/templates/kraken.provider.aws.tf.jinja2:  shared_credentials_file = "{{cluster.providerConfig.authentication.credentialsFile}}"
bash-4.3# 
```

**Comment by davidewatson** [comment]
Note that the `kraken.ssh` role is required for testing etcd health. So once my changes for #622 are finished and merged, the `kraken.provider.aws` role will start to fail if `~/.aws` is not used.

**Comment by coffeepac** [comment]
I can set AWS_SHARED_CREDENTIALS_FILE to make this work for the credentials file ([docs](http://boto3.readthedocs.io/en/latest/guide/configuration.html)) 

it looks like for #622 you can add a `-i cluster.nodepool.etcd.keyPair.privateKeyFile` to make that PR use the value set in the config file.

**Comment by coffeepac** [comment]
I can not use the environment variable AWS_SHARED_CREDENTIALS_FILE because that is for boto3 and ansible uses boto2 for some libraries, ie the ones I'm trying to make work here.  (https://stackoverflow.com/questions/41957949/is-boto3-supported-by-ansible). 

I am rather flummoxed.  Will attempt another workaround.

**Comment by coffeepac** [comment]
going to try adding a task to kraken.config that populates the accessKey and accessSecret if unset and credentialsFile is set.  nuisance.

**Comment by coffeepac** [comment]
resolved via:  #633.  Should have been closed when that was merged.
